server/src/main/java/com/linecorp/line/auth/fido/fido2/server/mds/MdsFetchTask.java

```diff
@@ -50,80 +32,61 @@
-    private static final String ALGORITHM_ES512 = "ES512";
 
-    private final MetadataRepository metadataRepository;
-    private final MetadataTocRepository metadataTocRepository;
     private final MdsProtocolClient mdsProtocolClient;
 
+    private final MdsV3MetadataHelper mdsV3MetadataHelper;
+
     private final Queue<MdsInfo> metadataSourceQueue = new LinkedList<>();
 
-    private boolean enableMds;
+    private final boolean enableMds;
 
     @Autowired
-    public MdsFetchTask(MdsConfig mdsConfig, MetadataRepository metadataRepository, MetadataTocRepository metadataTocRepository) {
-        this.metadataRepository = metadataRepository;
-        this.metadataTocRepository = metadataTocRepository;
+    public MdsFetchTask(MdsConfig mdsConfig, MdsV3MetadataHelper mdsV3MetadataHelper) {
         this.enableMds = mdsConfig.isEnableMds();
+        this.mdsV3MetadataHelper = mdsV3MetadataHelper;
 
         mdsProtocolClient = new MdsProtocolClient(mdsConfig.getSources().get(0).getEndpoint());
 
-        if(enableMds) {
+        if (enableMds) {
             metadataSourceQueue.addAll(mdsConfig.getSources());
         }
     }
 
-    public void refreshMetadata() {
+    private void refreshMetadata() {
         MdsInfo mdsInfo = metadataSourceQueue.poll();
         if (mdsInfo == null || !mdsInfo.isEnabled()) {
             return;
         }
         log.info("Handle MDS with following source: {}", mdsInfo);
-        String mdsSourceEndpoint = mdsInfo.getEndpoint();
-        if ("fido-mds-v2".equals(mdsInfo.getName()) && mdsInfo.getAccessToken() != null ) {
-            mdsSourceEndpoint += "?token=" + mdsInfo.getAccessToken();
-        } else if (mdsInfo.getName().contains("conformance")) {
-            mdsSourceEndpoint += mdsInfo.getAccessToken();
-        }
         log.info("Start fetching Metadata TOC");
-        mdsProtocolClient.fetchAsyncMetadataToc(mdsSourceEndpoint, new Callback<ResponseBody>() {
+        mdsProtocolClient.fetchAsyncMetadataToc(mdsInfo.getEndpoint(), new Callback<ResponseBody>() {
             @Override
             public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
                 if (response.isSuccessful()) {
                     log.debug("Metadata successfully fetched");
-                    String metadataToc;
                     String url = call.request().url().toString();
                     try {
                         if (response.body() == null) {
                             log.error("Metadata TOC fetch failed");
                             return;
                         }
-                        metadataToc = response.body().string();
+                        String metadataToc = response.body().string();
                         // handleMetadataToc
                         MetadataTOCResult result = handleMetadataToc(url, metadataToc, mdsInfo);
                         log.info("Metadata TOC from MDS ({}) handling result: {}", mdsInfo.getName(), result);
```

Comment:
This generate way too much `info` log and is not really helpful.
Maybe it can be downgraded to `debug`.